test/integration/gateway_test.go

```diff
@@ -63,7 +63,8 @@ func TestUnmanagedGatewayBasics(t *testing.T) {
 	require.Eventually(t, func() bool {
 		gw, err = gatewayClient.GatewayV1beta1().Gateways(ns.Name).Get(ctx, defaultGatewayName, metav1.GetOptions{})
 		require.NoError(t, err)
-		return gw.Annotations[annotations.GatewayClassUnmanagedAnnotation] == "kong-system/ingress-controller-kong-proxy"
+		return strings.Contains(gw.Annotations[annotations.GatewayClassUnmanagedAnnotation],
+			"kong-system/ingress-controller-kong-proxy")
```

Comment:
We could also verify if `kong-system/ingress-controller-kong-udp-proxy` is included.

Reply:
I omitted it here since the test seemed to be more geared towards confirming that the Gateway went through its initial bootstrap at all, i.e. checking to confirm that annotation is not nothing and also not complete nonsense, rather than checking for an exact set of all expected services.

The UDPRoute tests in turn cover the presence of the UDP service string (they'll fail if none is present).

Reply:
Alright, well, there are additional checks against the address set, and making those only check the one is annoying, so a9ccd73bf4fe7faad08b946d21e8459cba3838f9 does both services for both checks.

We _(heavily weighted) probably_ won't have ever-expanding publish services (at least not in test) beyond TCP and UDP, so testing exactly two is _probably_ fine.

There is arguably still upstream weirdness here where our status indicates N addresses as generally associated with the Gateway, but those addresses only serve a subset of the Listeners, with no indication which is which.